### PR TITLE
Fix bind problem for FrSky XM+ v2.1.2 firmware

### PR DESCRIPTION
**Pull request by MJ666**
Based on an update of the MPM code. All thanks go to Pascal for doing the fix. 

https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/issues/543

### CODE REVIEW

File: src/protocol/frskyx_cc2500.c [hunk 1]
```diff
@@ -199,16 +199,19 @@ static void frskyX_build_bind_packet()
             memcpy(&packet[17], (void *)"\x55\xAA\x5A\xA5", 4);   // CH9-16
     } else {
         // packet 1D 03 01 0E 1C 02 00 00 32 0B 00 00 A8 26 28 01 A1 00 00 00 3E F6 87 C7 00 00 00 00 C9 C9
-        packet[5] = 0x02;                   // Unknown but constant ID?
+        // Unknown bytes
+        if(state & 0x01)
+            memcpy(&packet[7],"\x00\x18\x0A\x00\x00\xE0\x02\x0B\x01\xD3\x08\x00\x00\x4C\xFE\x87\xC7",17);
+        else
+            memcpy(&packet[7],"\x27\xAD\x02\x00\x00\x64\xC8\x46\x00\x64\x00\x00\x00\xFB\xF6\x87\xC7",17);
```
Comment (PhracturedBlue) [inline src/protocol/frskyx_cc2500.c:206]: Lint Error:
```
203:  Missing space before ( in if(  
204:  Missing space after ,  
206:  Missing space after ,  
```

File: src/protocol/frskyx_cc2500.c [hunk 2]
```diff
@@ -199,16 +199,19 @@ static void frskyX_build_bind_packet()
             memcpy(&packet[17], (void *)"\x55\xAA\x5A\xA5", 4);   // CH9-16
     } else {
         // packet 1D 03 01 0E 1C 02 00 00 32 0B 00 00 A8 26 28 01 A1 00 00 00 3E F6 87 C7 00 00 00 00 C9 C9
-        packet[5] = 0x02;                   // Unknown but constant ID?
+        // Unknown bytes
+        if (state & 0x01)
+            memcpy(&packet[7], "\x00\x18\x0A\x00\x00\xE0\x02\x0B\x01\xD3\x08\x00\x00\x4C\xFE\x87\xC7",17);
+        else
+            memcpy(&packet[7], "\x27\xAD\x02\x00\x00\x64\xC8\x46\x00\x64\x00\x00\x00\xFB\xF6\x87\xC7",17);
```
Comment (PhracturedBlue) [inline src/protocol/frskyx_cc2500.c:206]: Lint Error:
```
204:  Missing space after ,  
206:  Missing space after ,  
```

**Comment by howels** [comment]
@MJ666 @TheRealMoeder since this PR merge nightly builds have been broken, please can you check what happened?

**Comment by MJ666** [comment]
Sorry dot have any control over the nightly build process. Was also asking already in the forum.

**Comment by howels** [comment]
> Sorry dot have any control over the nightly build process. Was also asking already in the forum.

Can you revert the PR and check that all build targets are working?

**Comment by MJ666** [comment]
As you may see this PR passed all checks and should build without issues. It also builds without issues in my private build environment. Anyhow I'm not able to revert this PR since I dont have the permission to do so.

**Comment by howels** [comment]
Any ideas why the devo6 image build failed? https://travis-ci.org/github/DeviationTX/deviation/builds/774621651

**Comment by howels** [comment]
Completely flawed - builds fine locally here for devo6 on the same Ubuntu 18.04 base which I think Travis is using.

**Comment by gdenton61** [comment]
My local build (on Ubuntu 20.04.3) the devo6 builds fine but the devo7e fails with the same error as shown on lines 1416 & 1417 of https://travis-ci.org/github/DeviationTX/deviation/builds/774621651 referenced above:

arm-none-eabi-ld: objs/devo7e/frskyx_cc2500.elf section `.bss' will not fit in region `ram'
arm-none-eabi-ld: region `ram' overflowed by 18 bytes

**Comment by MJ666** [comment]
Mhhh, it really looks to be the FrSky protocol does not fit anymore into memory of the low end Devo's. There was quite some optimization done for the v2.x protocol update to make it fit. I guess there are now no more other option for optimisation than excluding this fix for the Devo7 builds. Not sure why it builds successful during merge in the first place?

**Comment by howels** [comment]
Can we disable it for limited-flash targets for now in order to get nightly builds rolling again and optimise for space later if that requires time and research @MJ666 ?

**Comment by MJ666** [comment]
Added a new PR to apply the XM+ fix only to non modular targets. #1015

**Pull request merged**